src/test/scala/chisel3/tests/BundleLiteralsSpec.scala

```diff
@@ -36,35 +25,44 @@ class BundleLiteralSpec extends FlatSpec with ChiselScalatestTester {
       io.aOut := io.in.a
       io.bOut := io.in.b
     }) { c =>
-      c.io.in.poke(c.io.in.Lit(0.U, 1.U))
+      c.io.in.poke(c.io.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
       c.io.aOut.expect(0.U)
       c.io.bOut.expect(1.U)
 
-      c.io.in.poke(c.io.in.Lit(2.U, 5.U))
+      c.io.in.poke(c.io.in.cloneType.Lit(_.a -> 2.U, _.b -> 5.U))
       c.io.aOut.expect(2.U)
       c.io.bOut.expect(5.U)
     }
   }
 
   it should "expect Bundle literals" in {
     test(new PassthroughModule(new DoubleElements)) { c =>
-      c.in.poke(c.in.Lit(0.U, 1.U))
-      c.out.expect(c.in.Lit(0.U, 1.U))
-      c.in.poke(c.in.Lit(2.U, 5.U))
-      c.out.expect(c.in.Lit(2.U, 5.U))
+      c.in.poke(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
+      c.out.expect(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
+      c.in.poke(c.in.cloneType.Lit(_.a -> 2.U, _.b -> 5.U))
+      c.out.expect(c.in.cloneType.Lit(_.a -> 2.U, _.b -> 5.U))
     }
   }
 
   it should "fail on expect mismatch" in {
     assertThrows[exceptions.TestFailedException] {
       test(new PassthroughModule(new DoubleElements)) { c =>
-        c.in.poke(c.in.Lit(0.U, 1.U))
-        c.out.expect(c.in.Lit(0.U, 2.U))
+        c.in.poke(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
+        c.out.expect(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 2.U))
       }
     }
   }
 
-  ignore should "peek Bundle literals" in {
-    // TODO to be written
+  // peek on bundle not supported yet, this test will fail and should
+  // be altered when BundleLiteral peeking works
+  it should "not peek Bundle literals at this point in time" in {
+    intercept[LiteralTypeException] {
+      test(new PassthroughModule(new DoubleElements)) { c =>
+        c.in.poke(c.in.cloneType.Lit(_.a -> 0.U, _.b -> 1.U))
+        val output = c.out.peek()
+        output.a === 0.U should be(true.B)
```

Comment:
Uh, I really don't think that will work, since we don't do const prop in the frontend - it will just end up as a wire, and that's assuming you don't get some error for trying to invoke HCL operations outside a Builder context.

That being said, I'm not sure how we check for literal equality.

Reply:
@ducky64 Can you think of a better example of a use case for peeking a Bundle?

Reply:
Test is now an ignore